transcom/mymove
# B-21393 Integration Follow Up

## [B-21393](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21393)
### [Part 1](https://github.com/transcom/mymove/pull/13860)
## Summary

Update the allowable weight field on the PPM shipment to be used when calculating all PPM closeout values when the allowable weight is less than the net weight.

**In this PR for this backlog:**
- Updated the ppm_closeout and ppm_estimator services to use the shipment's allowable weight if that is less that the net weight.
- Removed unused and untested UI component for the SC to edit the net weight of a shipment (became unused in B-18163)
- Updated some test shipments that have been submitted for closeout to populate the allowable weight field.
- Fixed some pre-existing copy & paste issues and typos.

**In the Part 1 PR for this backlog:**
- Moved the allowable weight column from weight_tickets to ppm_shipments to that there is one consolidated allowable weight per shipment rather than one per trip.
- The initial allowable weight is now calculated when the customer submits their PPM for closeout rather than in the UI as the SC opens each weight ticket.
- The new allowable weight is editable by the SC when reviewing weight tickets.
- New allowable weight field is used to calculate GCC rather than the sum of the allowable weights of the shipment's weight tickets

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

To recreate the real-world scenario where these changes will be used, we'll be creating two PPMs whose total weight will exceed the service members weight allowance. On the second PPM the counselor will set the allowable weight so that `Shipment1.NetWeight + Shipment2.AllowableWeight <= ServiceMember.WeightAllowance`. Feel free to use slightly different values that those listed below as long as you end up in the same scenario. The values below are arbitrary.

1. Access the customer app, create an Air Force service member (branch probably shouldn't matter).
2. Use JB Charleston, SC 29404 as your current location.
3. Use Scott AFB, IL 62225 as your new location.
4. Select O-4 as your pay grade, save.
5. Click on set up your shipments.
6. You should have a weight allowance of 14,000lbs.
7. Create a new move with at least two PPMs. Both should have a origin zip of 29404 and a destination zip of 62225.
8. This shouldn't matter but I set my estimated weights to 7,000lbs each.
9. Submit both moves.
10. Login as a Services Counselor (AGFM) and approve the move.
11. Log back in as your original customer and upload weight tickets for both shipments. At least one of the shipments should have two weight tickets to verify the weights are consolidated accurately. I set one shipment to a net weight of 7,000lbs and the other to 8,000lbs. Submit both for closeout.
13. Log back in as a Services Counselor and review the documents for the 7,000lbs shipment. Approve the weight ticket(s). Take note (a screenshot) of of the Incentives/Costs and Incentive Factors values. Approve the shipment.
14. Review the documents for the 8,000lbs shipment. The default allowable weight should be 8,000lbs (the net weight). Set it to 7,000lbs (so the total across both shipments does not exceed the service members weight allowance).
15. Approve the weight ticket(s). Take note (a screenshot) of of the Incentives/Costs and Incentive Factors values. 
16. The values for both the 7,000lbs shipment and the 8,000lb shipment with a 7,000lbs allowable weight should be the same.

Optionally repeat theses steps on the latest main to verify the values you see are the same for a 8,000 and 7,000lbs shipments of the same distance on the same days.

To be added:
Screenshots of the values for identical shipment created in main to compare.

No UI changes in this follow up so no ANDI review will be necessary.

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
